### PR TITLE
feat(login): use default org for login without provided org context

### LINKED ISSUE #6567: Trigger Default Organization Settings
At the moment we trigger the instance settings when a user opens up the ZITADEL login. As a user is always bound to a organization, it doesn't really make sense to trigger the instance settings. 
Per default the settings (login and branding) of the default organization should be shown.

**Acceptance Critiera**
- [ ] Trigger login settings of the default organization
- [ ] Trigger branding settings of the default organization
- [ ] Users with other usernames should still be able to login they will be redirected to the next screen of their organization
- [ ] All settings should be taken from default org (password, domain settings)
- [ ] Restriction of users only happens when organization scope is sent
- [ ] This should be changed for newly setuped instances, for existing instances, we want to have it as it is for the moment. Maybe a setting where they can change to the new setting.

### PR DESCRIPTION
**Pull request by livio-a**
This PR will change the default context (e.g. policies, IDPs) for the login UI to use the default organisation rather than the instance.

This behaviour change is currently enabled through a feature flag on the instance.

New instances will have the flag enabled by default.

Existing instances can switch to the new default by activating the flag. IAM_OWNER can do this through the  corresponding endpoint in the Admin API. This flag change is irreversible!

closes #6567 

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/command/instance_feature_model.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/feature"
+)
+
+type FeatureWriteModel[T feature.SetEventType] struct {
+	eventstore.WriteModel
+
+	feature domain.Feature
+
+	Type T
```
Comment (stebenz) [inline internal/command/instance_feature_model.go:17]: Shouldn't this attribute be "Value" or something? This "Type" seems rather confusing.

File: internal/command/instance_feature_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		err     error
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		res    res
+	}{
+		{
+			"unknown feature",
```
Comment (stebenz) [inline internal/command/instance_feature_test.go:40]: Handling of feature with different type?

File: proto/zitadel/system.proto [hunk 3]
```diff
@@ -879,3 +891,17 @@ message FailedEvent {
     }
   ];
 }
+
+message SetInstanceFeatureRequest {
+  string instance_id = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
+  // the feature (id) according to [internal/domain/feature.go]
+  int32 feature_id = 2 [(validate.rules).int32 = {gt: 0}];
+  // value based on the feature type
+  oneof value {
```
[the remark this answers is not in the document]
Comment (livio-a) [reply proto/zitadel/system.proto:900]: validation added

File: internal/api/grpc/system/feature.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+func (s *Server) setInstanceFeature(ctx context.Context, req *system_pb.SetInstanceFeatureRequest) (*domain.ObjectDetails, error) {
+	feat := domain.Feature(req.FeatureId)
+	if !feat.IsAFeature() {
+		return nil, errors.ThrowInvalidArgument(nil, "SYST-SGV45", "Errors.Feature.NotExisting")
+	}
+	switch t := req.Value.(type) {
+	case *system_pb.SetInstanceFeatureRequest_Bool:
+		return s.command.SetBooleanInstanceFeature(ctx, feat, t.Bool)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/api/grpc/system/feature.go:30]: the type is now checked based on the feature when initialising the writemodel

File: internal/command/instance_feature_model.go (hunk 1 above)
Comment (livio-a) [reply internal/command/instance_feature_model.go]: renamed it in the write model and event type

File: internal/command/instance_feature_test.go (hunk 2 above)
Comment (livio-a) [reply internal/command/instance_feature_test.go:41]: tested with the changes of the type check itself

File: internal/api/grpc/admin/feature.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"context"
+
+	object_pb "github.com/zitadel/zitadel/internal/api/grpc/object"
+	"github.com/zitadel/zitadel/internal/domain"
+	admin_pb "github.com/zitadel/zitadel/pkg/grpc/admin"
+)
+
+func (s *Server) ActivateFeatureLoginDefaultOrg(ctx context.Context, _ *admin_pb.ActivateFeatureLoginDefaultOrgRequest) (*admin_pb.ActivateFeatureLoginDefaultOrgResponse, error) {
+	details, err := s.command.SetBooleanInstanceFeature(ctx, domain.FeatureLoginDefaultOrg, true)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/api/grpc/admin/feature.go:12]: as state in the description: the flag is the new default, current instances are only able to switch to the new default

**Review by stebenz — APPROVED** [review]

**Pull request merged**